src/App.jsx

```diff
@@ -19,12 +21,20 @@ export function App() {
 	 * to create and join a new list.
 	 */
 	const [listToken, setListToken] = useStateWithStorage(
-		'my test list',
+		null,
 		'tcl-shopping-list-token',
 	);
 
+	function handleClick() {
+		const token = generateToken();
+		setListToken(token);
+		navigateTo('/list');
+	}
+
 	useEffect(() => {
-		if (!listToken) return;
+		if (listToken) {
+			navigateTo('/list');
+		} else return;
```

Comment:
nice work on adding the conditional navigation here while still keeping the code simple and readable to return to avoid throwing errors in streamListItems when there is a missing token